mlflow/store/tracking/abstract_store.py

```diff
@@ -21,7 +23,7 @@ def __init__(self):
         Empty constructor for now. This is deliberately not marked as abstract, else every
         derived class would be forced to create one.
         """
-        pass
+        self._async_logging_queue = AsyncLoggingQueue(logging_func=self.log_batch)
```

Comment:
@chenmoneygithub @sagarsumant Can we avoid starting a logging loop background thread until the first time someone makes an async logging call? I worry about introducing an always-running background thread to existing workloads that don't do async logging